Persistence/Schema/MappingType.hs

```diff
@@ -0,0 +1,65 @@
+module Persistence.Schema.MappingType where
+
+import Data.List (isPrefixOf)
+import Database.Persist.TH
+
+data MappingType = LocalDef | LocalThmOpen | LocalThmProved
+                 | GlobalDef | GlobalThmOpen | GlobalThmProved
+                 | HidingDef
+                 | FreeDef | CofreeDef | NPFreeDef | MinimizeDef
+                 | HidingOpen | HidingProved
+                 | HidingFreeOpen | HidingCofreeOpen
+                 | HidingNPFreeOpen | HidingMinimizeOpen
+                 | HidingFreeProved | HidingCofreeProved
+                 | HidingNPFreeProved | HidingMinimizeProved
```

Comment:
these names should be changed: the "Hiding" needs to be removed from the last 8 constructors. See also https://github.com/ontohub/ontohub-backend/blob/add_graphql_types_for_document_data/app/graphql/types/mapping_type_enum.rb

Reply:
But they are constructed in LibEnv from HidingFreeOrCofreeThm and not FreeOrCofreeDefLink. See https://github.com/spechub/Hets/blob/9c217c0bf6d4b39f9f5eb721e19a9a3125998d60/Persistence/DevGraph.hs#L1114-L1127